x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/parser/AbstractBuilder.java

```diff
@@ -137,50 +133,76 @@ public static String unquoteString(Source source) {
         checkForSingleQuotedString(source, text, 0);
 
         text = text.substring(1, text.length() - 1);
-        StringBuffer resultString = new StringBuffer();
-        Matcher regexMatcher = slashPattern.matcher(text);
-
-        while (regexMatcher.find()) {
-            String group = regexMatcher.group();
-            String replacement;
-
-            switch (group) {
-                case "\\t":
-                    replacement = "\t";
-                    break;
-                case "\\b":
-                    replacement = "\b";
-                    break;
-                case "\\f":
-                    replacement = "\f";
-                    break;
-                case "\\n":
-                    replacement = "\n";
-                    break;
-                case "\\r":
-                    replacement = "\r";
-                    break;
-                case "\\\"":
-                    replacement = "\"";
-                    break;
-                case "\\'":
-                    replacement = "'";
-                    break;
-                case "\\\\":
-                    // will be interpreted as regex, so we have to escape it
-                    replacement = "\\\\";
-                    break;
-                default:
-                    // unknown escape sequence, pass through as-is
-                    replacement = group;
+        StringBuilder sb = new StringBuilder();
+
+        for (int i = 0; i < text.length();) {
+            if (text.charAt(i) == '\\') {
+                switch (text.charAt(++i)) {
```

Comment:
optional, if editing the PR again: it might be worth adding a comment noting that the grammar makes the incrementing safe (i.e. `text` will can never end with `\`)

Reply:
Sure, will do!